File: backend/src/Shalom.Application/Contracts/TodayPeopleDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// The connection slice of the Today aggregate. At most ONE nudge per day,
/// and none at all once any contact was recorded today — relationships are
/// invitations, never obligations (and never streaks).
/// </summary>
public record TodayPeopleDto(
    ConnectionNudgeDto? Nudge,
    IReadOnlyList<UpcomingDateDto> UpcomingDates,
    PrayerFocusDto PrayerFocus);

File: frontend/projects/api/src/lib/models/today-people-dto.ts
import { ConnectionNudgeDto } from './connection-nudge-dto';
import { PrayerFocusDto } from './prayer-focus-dto';
import { UpcomingDateDto } from './upcoming-date-dto';

/**
 * Connection slice of the Today aggregate. At most ONE nudge per day, and
 * none at all once any contact was recorded today. Streaks never apply
 * here — relationships are invitations, not obligations. `prayerFocus` is
 * the morning ritual's deterministic daily focus (always present).
 */
export interface TodayPeopleDto {
  readonly nudge: ConnectionNudgeDto | null;
  readonly upcomingDates: UpcomingDateDto[];
  readonly prayerFocus: PrayerFocusDto;
}
